src/fides/api/ops/schemas/privacy_experience.py

```diff
@@ -12,86 +12,172 @@
     """
 
-    disabled: Optional[bool] = False
+    acknowledgement_button_label: Optional[SafeStr]
+    banner_title: Optional[SafeStr]
+    banner_description: Optional[SafeStr]
     component: Optional[ComponentType]
-    delivery_mechanism: Optional[DeliveryMechanism]
-    regions: Optional[conlist(PrivacyNoticeRegion, min_items=1)]  # type: ignore
     component_title: Optional[SafeStr]
     component_description: Optional[SafeStr]
-    banner_title: Optional[SafeStr]
-    banner_description: Optional[SafeStr]
-    link_label: Optional[SafeStr]
     confirmation_button_label: Optional[SafeStr]
+    delivery_mechanism: Optional[DeliveryMechanism]
+    disabled: Optional[bool] = False
+    is_default: Optional[bool] = False
+    link_label: Optional[SafeStr]
     reject_button_label: Optional[SafeStr]
-    acknowledgement_button_label: Optional[SafeStr]
 
-    class Config:
-        """Populate models with the raw value of enum fields, rather than the enum itself"""
 
-        use_enum_values = True
-        orm_mode = True
-        extra = Extra.forbid
+class ExperienceConfigCreate(ExperienceConfigSchema):
+    """
+    An API representation to create ExperienceConfig.
+    This model doesn't include an `id` so that it can be used for creation.
+    It also establishes some fields _required_ for creation
+    """
+
+    component: ComponentType
+    delivery_mechanism: DeliveryMechanism
+    regions: List[PrivacyNoticeRegion]
+    component_title: SafeStr
 
     @validator("regions")
     @classmethod
     def validate_regions(
         cls, regions: List[PrivacyNoticeRegion]
     ) -> List[PrivacyNoticeRegion]:
-        """Assert regions aren't duplicated.  Without this, duplications get flagged as misleading duplicate data uses"""
+        """Assert regions aren't duplicated."""
         if len(regions) != len(set(regions)):
             raise ValueError("Duplicate regions found.")
         return regions
 
-
-class PrivacyExperienceCreate(PrivacyExperience):
+    @root_validator
+    def validate_attributes(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Validate minimum set of required fields exist given the type of component and/or delivery_mechanism"""
+        component: Optional[ComponentType] = values.get("component")
+        delivery_mechanism: Optional[DeliveryMechanism] = values.get(
+            "delivery_mechanism"
+        )
+
+        if delivery_mechanism == DeliveryMechanism.link and not values.get(
+            "link_label"
+        ):
+            raise ValueError(
+                "Link label required when the delivery mechanism is of type link."
+            )
+
+        if component == ComponentType.overlay:
+            if delivery_mechanism == DeliveryMechanism.banner:
+                required_banner_fields = [
+                    "acknowledgement_button_label",
+                    "banner_title",
+                    "confirmation_button_label",
+                    "reject_button_label",
```

Comment:
I included all three of these button labels here as "required" for setting up a banner overlay.  There may not be notice only notices, but this prevents us from having to check that. I'm trying to minimize the number of things we need to check on a notice when we create an ExperienceConfig and link a region to it.